### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by bircni**
Before I start: Very cool project and cool idea!!!

here are some improvements - I also added some clippy lints and fixed them
It would be very important to remove all unwraps and handle Results with anyhow (my suggestion)
I also added some todos which could be done in the future

### CODE REVIEW

File: src/app/top_bottom.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                                }
+                                1 => &path,
+                                _ => {
+                                    path += e.to_str().unwrap();
+                                    path.push(std::path::MAIN_SEPARATOR);
+                                    e.to_str().unwrap()
+                                }
+                            };
+                            if ui.button(text).clicked() {
+                                new_path = Some(path.into());
```
Comment (Leinnan) [inline src/app/top_bottom.rs]:
```suggestion
                                *new_path = Some(path.into());
```

File: src/app/top_bottom.rs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                                        e.to_str().unwrap().chars().rev().take(2).collect();
+                                    path += &last_two_chars.chars().rev().collect::<String>();
+                                    path.push(std::path::MAIN_SEPARATOR);
+                                    continue;
+                                }
+                                1 => &path,
+                                _ => {
+                                    path += e.to_str().unwrap();
+                                    path.push(std::path::MAIN_SEPARATOR);
+                                    e.to_str().unwrap()
```
Comment (Leinnan) [inline src/app/top_bottom.rs]:
```suggestion
                                    let Some(s) = e.to_str() else {
                                        return;
                                    };
                                    path += s;
                                    path.push(std::path::MAIN_SEPARATOR);
                                    s
```

File: src/app/top_bottom.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                    let mut path: String = String::new();
+
+                    #[allow(unused_variables)] // todo fix unwraps here
+                    for (i, e) in self.cur_path.iter().enumerate() {
+                        #[cfg(windows)]
+                        {
+                            let text = match &i {
+                                0 => {
+                                    let last_two_chars: String =
+                                        e.to_str().unwrap().chars().rev().take(2).collect();
```
Comment (Leinnan) [inline src/app/top_bottom.rs]:
```suggestion
                                    let Some(s) = e.to_str() else {
                                        continue;
                                    };
                                    let last_two_chars: String = s.chars().rev().take(2).collect();
```

**Review by Leinnan — CHANGES_REQUESTED** [review]
First - great work! 
I have just few small improvements to make the code work on Windows as well and pass clippy there.

**Comment by bircni** [comment]
Shockers be fixed now! 😃

**Comment by Leinnan** [comment]
Once again, great work! Thanks!

**Review by Leinnan — APPROVED** [review]

**Pull request merged**